### PR TITLE
techdebt/MTSDK-541_no_feedback_when_detach_is_called_with_an_invalid_id

### PR DESCRIPTION
**Pull request by ksw5**
- Logs an error when detach is called with an invalid attachment ID.

- Update `when detach() on not processed attachment` test.

### CODE REVIEW

File: transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/AttachmentHandlerImpl.kt [hunk 1]
```diff
@@ -85,6 +85,11 @@ internal class AttachmentHandlerImpl(
     }
 
     override fun detach(token: String, attachmentId: String): DeleteAttachmentRequest? {
```
Comment (AfanasievAnton) [inline transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/AttachmentHandlerImpl.kt:87]: - Add @Throws annotation to detach() function. 
- Also update the @Throws annotation of the MessagingClient.detach() as right now it's signature say that it throws ONLY IllegalStateException
- The comment is redundant.
- Create and use ErrorMessage.detachFailed() as the message parameter when you throw IllegalArgumentException instead of String. Reuse it later on in your unit test.

File: transport/src/androidUnitTest/kotlin/transport/core/AttachmentHandlerImplTest.kt [hunk 2]
```diff
@@ -293,12 +293,14 @@ internal class AttachmentHandlerImplTest {
 
     @Test
     fun `when detach() on not processed attachment`() {
-        val result = subject.detach(TestValues.Token, "not processed attachment id")
+        val invalidAttachmentId = "not processed attachment id"
```
Comment (AfanasievAnton) [inline transport/src/androidUnitTest/kotlin/transport/core/AttachmentHandlerImplTest.kt:296]: - Rename invalidAttachmentId to givenAttachmentId to indicate that this is an input value we use for testing.
- Use TestValues.DEFAULT_STRING instead of hardcoded  "not processed attachment id".
- I would like to see a unit test in MCAttachmentTests ->  `when detach() non existing attachmentId` updated to reflect the change. Initial assumption of this test was that when detach non existing attachment attachmentHandler will return null. But now, it throws an exception.

**Review by AfanasievAnton — CHANGES_REQUESTED** [review]

**Review by AfanasievAnton — APPROVED** [review]
Nice and clean!
